apps/rebar/src/rebar_prv_common_test.erl

```diff
@@ -307,10 +307,6 @@ select_tests(State, ProjectApps, CmdOpts, CfgOpts) ->
     Configs = lists:flatmap(fun(Filename) ->
                                 rebar_file_utils:consult_config(State, Filename)
                             end, SysConfigs),
-    %% NB: load the applications (from user directories too) to support OTP < 17
-    %% to our best ability.
-    rebar_paths:set_paths([deps, plugins], State),
```

Comment:
This is specifically not something that should be undone because it ensures that if there is a path clash across dependencies and plugins (because a plugin uses a dep from a different version from the main application), the dependency gets loaded at a higher priority than the plugin.

Reply:
Not sure if it's needed here without application loading. There's the same call in `do/2` just before running test suites.